src/server/scripts/Kalimdor/TempleOfAhnQiraj/boss_cthun.cpp

```diff
@@ -342,10 +342,10 @@ class boss_eye_of_cthun : public CreatureScript
                         if (DarkGlareTickTimer <= diff)
                         {
                             //Set angle and cast
-                            if (ClockWise)
-                                me->SetOrientation(DarkGlareAngle + DarkGlareTick * float(M_PI) / 35);
-                            else
-                                me->SetOrientation(DarkGlareAngle - DarkGlareTick * float(M_PI) / 35);
+                            float angle = (ClockWise) ? DarkGlareAngle + DarkGlareTick * float(M_PI) / 35 : DarkGlareAngle - DarkGlareTick * float(M_PI) / 35;
+
+                            me->SetOrientation(angle);
+                            me->SetFacingTo(angle);
```

Comment:
I wonder if both are needed or if SetOrientation is not required when calling SetFacingTo. Could you check what happens when not calling SetOrientation ? I'll ask around too.

Reply:
The direction of the spell cast isn't changed for some reason.

Reply:
Well, I have no idea if all SetOrientation() calls need a SetFacingTo() and if all SetFacingTo() need a SetOrientation().

These functions are still affected by some bugs with workarounds added years ago and never properly fixed, like https://github.com/TrinityCore/TrinityCore/commit/76667e45a902f86bd341d40efa28037137e5e541 . Maybe we should call SetOrientation() in SetFacingTo() by default ? Or maybe we shouldn't ?

We leave that discussion for another day

Reply:
Mostly another hackfix, but I think we should call SetFacingTo() in SetOrientation() here:
[Position.h](https://github.com/TrinityCore/TrinityCore/blob/e8849ca5a14719e7d395db7907bac26705f25dbf/src/server/game/Entities/Object/Position.h#L71)